### PR TITLE
TRUNK-5472: Added Test units for Allergy.hasSameValues And also Refact…

### PR DESCRIPTION
**Pull request by mozzy11**
Ticket : https://issues.openmrs.org/browse/TRUNK-5472

Added Test units for Allergy.hasSameValues And also Refactored it

### CODE REVIEW

File: api/src/test/java/org/openmrs/AllergyTest.java [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+	@Test(expected = NullPointerException.class)
+	public void hasSameValues_otherAllergyIsNull_shouldThrowNullPointerException() {
+		Allergy allergy = new Allergy();
+		allergy.hasSameValues(null);
+	}
+
+	@Test
+	public void hasSameValues_checkingAgainstSameInstanceOfAllergy_shouldInterpretAsSameValues() {
+		Allergy allergy = allergy();
+		allergy.hasSameValues(allergy());
```
Comment (HelioStrike) [inline api/src/test/java/org/openmrs/AllergyTest.java]: Shouldn't this line be AssertTrue(allergy.hasSameValues(allergy())); ?

File: api/src/test/java/org/openmrs/AllergyTest.java [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+	public void hasSameValues_allergyWithIdAndAllergyWithNullId_shouldBeInterpretAsNotSameValues() {
+		assertThat(allergyWithId(1).hasSameValues(allergyWithId(null)), is(false));
+	}
+
+	@Test
+	public void hasSameValues_allergiesWithDifferentPatientIds_shouldNotBeInterpretAsNotSameValues() {
+		Allergy allergy = allergyWithPatientId(1);
+		Allergy otherAllergy = allergyWithPatientId(2);
+
+		assertThat(allergy.hasSameValues(otherAllergy), is(false));
```
Comment (haripriya999) [inline api/src/test/java/org/openmrs/AllergyTest.java]: hasSameValues_allergyWithNullIdAndAllergyWithId_shouldBeInterpretAsNotSameValues() and hasSameValues_allergiesWithDifferentPatientIds_shouldNotBeInterpretAsNotSameValues() has logically the same code though the name of the functions refer completely opposite to each other. I suggest either to make changes to the 2nd function or have it removed.

File: api/src/test/java/org/openmrs/AllergyTest.java (hunk 1 above)
Comment (mozzy11) [reply api/src/test/java/org/openmrs/AllergyTest.java]: ya yor actually right. i just missed it out . thanks @HelioStrike

File: api/src/test/java/org/openmrs/AllergyTest.java (hunk 2 above)
Comment (mozzy11) [reply api/src/test/java/org/openmrs/AllergyTest.java]: alryt @haripriya999 , ill look at it

**Comment by gitcliff** [comment]
@mozzy11  am seeing a PR quality review error, is it an issue of concern or?

**Comment by mozzy11** [comment]
your right @gitcliff . the codacy quality is complaining about the test unit that misses an assert statement , which exactly @HelioStrike  mentioned about. am going to correct it. thanks

**Comment by mozzy11** [comment]
i upadted the PR and update @HelioStrike and @gitcliff 's  concerns

File: api/src/test/java/org/openmrs/AllergyTest.java (hunk 2 above)
Comment (mozzy11) [reply api/src/test/java/org/openmrs/AllergyTest.java]: no @haripriya999 , they are completey different logically.
with   hasSameValues_allergyWithNullIdAndAllergyWithId_shouldBeInterpretAsNotSameValues() 
the Allery =allergyWithId(*)

yet wtih  hasSameValues_allergiesWithDifferentPatientIds_shouldNotBeInterpretAsNotSameValues() 
the Allergy = allergyWithPatientId(*);

allergyWithPatientId(patient id) and  allergyWithId(id) are two different functions

Comment (haripriya999) [reply api/src/test/java/org/openmrs/AllergyTest.java]: Oh sorry, my bad! That part of code seemed a bit unclear to me :)

File: api/src/main/java/org/openmrs/Allergy.java [hunk 3]
```diff
@@ -247,50 +247,64 @@ public Date getDateLastUpdated() {
...
 		}
-		if (!OpenmrsUtil.nullSafeEquals(getComment(), allergy.getComment())) {
-			return false;
-		}
-		return hasSameReactions(allergy);
+		return true;
 	}
+
+	private boolean hasSameComment(Allergy allergy) {
+		return OpenmrsUtil.nullSafeEquals(getComments(), allergy.getComment());
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/Allergy.java:305]: Shouldn't this be:

```suggestion
		return OpenmrsUtil.nullSafeEquals(getComments(), allergy.getComments());
```

or

```suggestion
		return OpenmrsUtil.nullSafeEquals(getComment(), allergy.getComment());
```
?

**Comment by mozzy11** [comment]
cc @dkayiwa  @ibacher

**Pull request closed without merging**